### PR TITLE
Updated package design

### PR DESCRIPTION
**Pull request by kels271828**
The main thing to review in this PR is the new file structure.

### CODE REVIEW

File: src/onemod/stage/model_stage/kreg_stage.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+"""Kreg stage."""
+
+import fire
+
+from onemod.config.model_config import KregConfig
+from onemod.stage import ModelStage
+
+
+class KregStage(ModelStage):
```
Comment (blsmxiu47) [inline src/onemod/stage/model_stage/kreg_stage.py:9]: The way these stage classes are written means we can immediately write simple tests for the construction and running of a dummy Pipeline 👍 which is not a bad place to get things started I think

File: src/onemod/pipeline.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                stage.create_stage_params()
+        self._stages[stage.name] = stage
+
+    def _stage_from_json(self, stage: str, filepath: Path | str) -> Stage:
+        """Load stage object from JSON."""
+        if stage in self.stages:
+            raise ValueError(f"stage '{stage}' already exists")
+        with open(filepath, "r") as f:
+            stage_json = json.load(f)
+        if stage_type := stage_json["type"] in STAGE_DICT:
```
Comment (blsmxiu47) [inline src/onemod/pipeline.py:99]: I might need some convincing on the use of `STAGE_DICT` here. I'm not exactly sure what a best alternative looks like at this moment, but for some reason I feel like within the core Pipeline component there might be a nicer alternative to relying on a static dictionary of stage types. Might be missing something though

**Comment by blsmxiu47** [comment]
Thought on structure/naming - I wonder if we could make the names of directories which hold classes for implementations on top of the base `Stage` classes plural (i.e. `model_stage` -> `model_stages`, etc.). Tiny thing but to me that makes it a bit clearer that these are instances of Stages built on top of `ModelStage` (etc) rather than components required in the core implementation of `ModelStage` (etc)

File: src/onemod/stage/model_stage/kreg_stage.py (hunk 1 above)
Comment (geoffreyphipps-uw) [reply src/onemod/stage/model_stage/kreg_stage.py:9]: I think it is a good place to start :-)

**Review by blsmxiu47 — APPROVED** [review]
This will serve as a great base for us to work off of. Thanks for your hard work!

File: src/onemod/pipeline.py (hunk 2 above)
Comment (kels271828) [reply src/onemod/pipeline.py]: Agreed, removed for now; we can figure out an easier way to load built-in stages later

**Pull request merged**